### PR TITLE
add mongodb

### PR DESCRIPTION
**Pull request by halp1**
Also allowed scrolling for small windows on /[testament] and added  favicon.ico

### CODE REVIEW

File: lib/mongodb.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+async function connectToDatabase() {
+  if (cachedClient && cachedDb) {
+    return { client: cachedClient, db: cachedDb }
+  }
+
+  const client = await MongoClient.connect(uri, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
```
Comment (leo6liu) [inline lib/mongodb.js]: Why use these two settings?

File: lib/mongodb.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import { MongoClient } from 'mongodb'
```
Comment (leo6liu) [inline lib/mongodb.js:3]: Could you add a URL at the top of this file in a comment for where this code is from?

```
// code taken from: https://dogland.com

import { MongoClient } ...
```

File: .gitignore [hunk 3]
```diff
@@ -26,9 +26,12 @@ yarn-error.log*
 
 # local env files
 .env.local
-.env.development.local
```
Comment (leo6liu) [inline .gitignore]: don't remove .env.development.local from the gitignore. The reason it's there is so security things like usernames and passwords do not make their way into public repositories.

File: .env.development.local [hunk 4]
```diff
@@ -0,0 +1,2 @@
+MONGODB_URI=mongodb+srv://readUser:PEIiFE7YuQ5vGr6u@cluster0.cq1ho.mongodb.net/data?retryWrites=true&w=majority
```
Comment (leo6liu) [inline .env.development.local]: see comment in .gitignore this type of file should not appear in the public repo.

You can put this in the google drive.

File: lib/mongodb.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  const db = await client.db(dbName)
+
+  cachedClient = client
+  cachedDb = db
+
+  return { client, db }
+}
+
+export default async function mongo(collection, find = {}) {
```
Comment (leo6liu) [inline lib/mongodb.js:41]: Did you add this function?

File: next.config.js [hunk 6]
```diff
@@ -1,3 +1,3 @@
 module.exports = {
   reactStrictMode: true,
```
Comment (leo6liu) [inline next.config.js]: Why do we need this?

File: pages/[testament]/index.js [hunk 7]
```diff
@@ -3,91 +3,30 @@ import Link from 'next/link';
 import { useRouter } from 'next/router';
 import { motion } from 'framer-motion';
 import Book from '/components/Book';
+import mongo from '/lib/mongodb';
+import './scrollbar.css';
```
Comment (leo6liu) [inline pages/[testament]/index.js]: can we inline this CSS? it seems to be used only once right?

File: pages/[testament]/index.js [hunk 8]
```diff
@@ -3,91 +3,30 @@ import Link from 'next/link';
...
 // TODO: add description
-export default function Page() {
+export default function Page({ booksData }) {
 
   // get testament from router
   const router = useRouter();
   const { testament } = router.query;
-  const initialWidth = (testament === 'ot') ? '14rem' : '10rem';
+
+	if (router.isFallback) {
```
Comment (leo6liu) [inline pages/[testament]/index.js:15]: tabbing looks weird.

This needs a comment.

File: pages/[testament]/index.js [hunk 9]
```diff
@@ -120,15 +59,15 @@ export default function Page() {
           </Link>
         </div>
 
-        <div className="flex flex-row justify-center items-center py-16 px-8 gap-6">
+        <div className="flex flex-row justify-center items-center py-16 px-8 gap-6" style = {{ overflow: 'scroll' }}>
```
Comment (leo6liu) [inline pages/[testament]/index.js]: why not use the tailwind class?

File: pages/[testament]/index.js [hunk 10]
```diff
@@ -120,15 +59,15 @@ export default function Page() {
...
           <motion.div
             initial="together" animate="apart"
             variants={shelfVariants} transition={{ duration: 1.5 }}
-            className="flex flex-row justify-start h-80 rounded-3xl"
+            className="flex flex-row justify-start h-80 rounded-3xl w-48 force-overflow"
```
Comment (leo6liu) [inline pages/[testament]/index.js]: see comment about the css import

File: pages/[testament]/index.js [hunk 11]
```diff
@@ -145,4 +84,32 @@ export default function Page() {
       </main>
     </>
   );
-}
\ No newline at end of file
+}
+
+export async function getStaticPaths() {
+	const data = await mongo('books');
```
Comment (leo6liu) [inline pages/[testament]/index.js:89]: inconsistent blank lines... stick to a pattern

File: pages/[testament]/index.js [hunk 12]
```diff
@@ -145,4 +84,32 @@ export default function Page() {
...
+	console.log(paths);
+	
+	return {
+		paths: paths,
+		fallback: false,
+	};
+}
+
+export async function getStaticProps() {
+
```
Comment (leo6liu) [inline pages/[testament]/index.js:107]: inconsistent blank lines... stick to a pattern

File: pages/[testament]/scrollbar.css [hunk 13]
```diff
@@ -0,0 +1,7 @@
+.force-overflow {
```
Comment (leo6liu) [inline pages/[testament]/scrollbar.css]: see comment in pages/[testament]/index.js

**Review by leo6liu — COMMENTED** [review]
see comments 🥝

File: lib/mongodb.js (hunk 5 above)
Comment (halp1) [reply lib/mongodb.js:41]: yes
Makes stuff easier :)

File: pages/[testament]/index.js (hunk 7 above)
Comment (halp1) [reply pages/[testament]/index.js]: I was going to put the scrollbar css in there too. You can't do that with inline css in react/next without it getting really messy. (doesn't look like css at all)

This does need to be moved to the _app.js file though.

File: next.config.js (hunk 6 above)
Comment (halp1) [reply next.config.js]: I didn't change that...
Git got confused
That is part of the default next config file

File: pages/[testament]/index.js (hunk 7 above)
Comment (halp1) [reply pages/[testament]/index.js]: Also, framer motion changes the style attribute, so you have to have an !important to force past that.

File: next.config.js (hunk 6 above)
Comment (leo6liu) [reply next.config.js]: ahh i understand. I being a dummy.

File: lib/mongodb.js (hunk 5 above)
Comment (leo6liu) [reply lib/mongodb.js:41]: okay. I'll look at it more closely and think about naming. Right now it is not descriptive enough and there is no commenting telling what it does either.

File: lib/mongodb.js (hunk 1 above)
Comment (halp1) [reply lib/mongodb.js]: Following the tutorial

File: pages/[testament]/index.js (hunk 7 above)
Comment (leo6liu) [reply pages/[testament]/index.js]: maybe put it in the global css and give it a very descriptive comment explaining why it's there and where it is being used.

File: lib/mongodb.js (hunk 2 above)
Comment (leo6liu) [reply lib/mongodb.js:3]: i can't find this file at that URL. All I can find is this: https://github.com/vercel/next.js/blob/canary/examples/with-mongodb/lib/mongodb.js

File: pages/[testament]/index.js [hunk 14]
```diff
@@ -145,4 +83,31 @@ export default function Page() {
       </main>
     </>
   );
-}
\ No newline at end of file
+}
+
+export async function getStaticPaths() {
```
Comment (leo6liu) [inline pages/[testament]/index.js:88]: Why do these two functions need to be exports?

Also put a comment above each of the two explaining what their purpose is. Like what do they fetch for us?

**Pull request merged**